src/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs

```diff
@@ -38,5 +38,21 @@ public static class TaskTimeoutExtensions
                 throw new TimeoutException();
             }
         }
+
+        public static async Task<bool> WhenAllWithTimeout(this Task[] tasks, int millisecondsTimeout)
+        {
+            var cts = new CancellationTokenSource();
+            Task task = Task.WhenAll(tasks);
+            if(task == await Task.WhenAny(task, Task.Delay(millisecondsTimeout, cts.Token)))
+            {
+                cts.Cancel();
+                await task;
```

Comment:
Here it actually would be good to use `ConfigureAwait(false)`.  See my similar comment in your System.Net.Security PR.  This could also just use whatever helpers you put in Common rather than duplicating it.